vine/src/parser.rs

```diff
@@ -248,10 +248,24 @@ impl<'ctx, 'src> VineParser<'ctx, 'src> {
 
   fn _parse_expr_prefix(&mut self, span: usize) -> Parse<'src, ExprKind> {
     if self.eat(Token::Return)? {
+      if self.check(Token::Semi) {
+        self.eat(Token::Semi)?;
+        return Ok(ExprKind::Return(Box::new(Expr {
+          span: self.span(),
+          kind: ExprKind::Tuple(Vec::new()),
+        })));
+      }
       return Ok(ExprKind::Return(Box::new(self.parse_expr_bp(BP::ControlFlow)?)));
     }
     if self.eat(Token::Break)? {
-      return Ok(ExprKind::Break);
+      if self.check(Token::Semi) {
+        self.eat(Token::Semi)?;
+        return Ok(ExprKind::Return(Box::new(Expr {
+          span: self.span(),
```

Comment:
2 things feel off here:
- the span
- check() then eat()

Reply:
`eat` returns a boolean, so it could just be `if self.eat(Token::Semi)? { ... }`. though for this specific case, I don't think it should eat the `;`, as this would make something like `break; + 1` a valid expression; it should leave the `;` there to be processed by `parse_stmt`

Reply:
And yeah, the span is strange here. I think it might be cleaner here for it to be an `Option<Expr>`, rather than using `()` like I suggested.